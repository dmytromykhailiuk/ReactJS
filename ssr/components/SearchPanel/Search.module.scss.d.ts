// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'search': string;
  'search__button': string;
  'search__input': string;
  'search__title': string;
}
export const cssExports: CssExports;
export default cssExports;
