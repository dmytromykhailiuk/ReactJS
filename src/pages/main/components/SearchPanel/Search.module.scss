@import '../../../../scss/mixins';

.search {
  display: block;
  width: 88%;
  margin: auto;

  &__button, &__input {
    margin: auto;
  }

  &__button {
    display: flex;
    margin-top: 7vh;
    justify-content: center;
  }

  @include medium-screen {
    display: flex;
    width: 80%;

    &__input {
      margin-right: 20px;
      flex-grow: 1;
    }

    &__button {
      margin-top: 0;
    }
  }
}