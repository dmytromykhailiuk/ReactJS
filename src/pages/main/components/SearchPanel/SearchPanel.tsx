import React from 'react';
import { Button, Input } from "shared/components";
import { ButtonTypes } from "shared/enums";
import { useInputValue } from 'shared/hooks';
import classes from "./Search.module.scss";

interface SearchPanelProps {
  onChangeSearchingValue?: (searchingValue: string) => void;
}

const SearchPanel: React.FC<SearchPanelProps> = ({ onChangeSearchingValue = () => {} }) => {
  const {value, onChangeValue} = useInputValue('');
  
  return (
    <div className={classes.search}>
      <div className={classes.search__input}>
        <Input 
          value={value} 
          placeholder="What do you want to watch?"
          onChange={onChangeValue}
        />
      </div>
      <div className={classes.search__button}>
        <Button 
          type={ButtonTypes.PRIMARY} 
          onButtonClicked={() => onChangeSearchingValue(value)} 
        >
          SEARCH
        </Button>
      </div>
    </div>
  )
};

export default SearchPanel;
