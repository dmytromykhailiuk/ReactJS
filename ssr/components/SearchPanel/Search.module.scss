@import '../../styles/mixins';

.search {
  display: block;
  width: 88%;
  margin: auto;

  &__button,
  &__input {
    margin: auto;
  }

  &__title {
    margin: 40px 10%;
    font-size: 40px;
    color: $white-color;
    font-weight: normal;
    text-align: center;
    letter-spacing: 5px;
    text-shadow: 0 0 1px $dark-grey-color;

    @include medium-screen {
      margin: 80px 10%;
      font-size: 55px;
      text-align: left;
    }
  }

  &__button {
    display: flex;
    margin-top: 7vh;
    justify-content: center;
  }

  @include medium-screen {
    display: flex;
    width: 80%;

    &__input {
      margin-right: 20px;
      flex-grow: 1;
    }

    &__button {
      margin-top: 0;
    }
  }
}