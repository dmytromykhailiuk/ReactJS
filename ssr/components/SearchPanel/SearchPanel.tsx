import React, { useCallback } from 'react';
import { Button, Input } from '../../shared/components';
import { ButtonTypes } from '../../shared/enums';
import { useInputValue } from '../../shared/hooks';
import classes from './Search.module.scss';
import { useDispatch } from 'react-redux';
import { MoviesAction } from '../../store';

const SearchPanel = () => {
  const { value, onChangeValue } = useInputValue('');
  const dispatch = useDispatch();

  const onSubmit = useCallback(
    (event: React.SyntheticEvent) => {
      event.preventDefault();
      dispatch(MoviesAction.searchMoviesAction({ searchingValue: value, scrollToMovies: () => {} }));
    },
    [value],
  );

  return (
    <>
      <h1 className={classes.search__title}>FIND YOUR MOVIE</h1>
      <form className={classes.search} onSubmit={onSubmit}>
        <div className={classes.search__input}>
          <Input value={value} name="search" placeholder="What do you want to watch?" onChange={onChangeValue} />
        </div>
        <div className={classes.search__button}>
          <Button type={ButtonTypes.PRIMARY} isSubmit={true}>
            SEARCH
          </Button>
        </div>
      </form>
    </>
  );
};

export default SearchPanel;
