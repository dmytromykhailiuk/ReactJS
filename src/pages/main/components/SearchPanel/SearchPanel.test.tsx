import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import SearchPanel from './SearchPanel';

describe('MovieList', () => {
  it('should match snepshot', () => {
    const wrapper = shallow(<SearchPanel onChangeSearchingValue={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call 'onChangeSearchingValue' with '' when submit", () => {
    const onChangeSearchingValue = jest.fn();
    const preventDefault = jest.fn();
    const wrapper = shallow(<SearchPanel onChangeSearchingValue={onChangeSearchingValue} />);
    wrapper.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalled();
    expect(onChangeSearchingValue).toHaveBeenCalledWith('');
  });
});
