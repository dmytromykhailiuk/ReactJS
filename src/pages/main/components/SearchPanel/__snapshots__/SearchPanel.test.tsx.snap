// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should match snepshot 1`] = `
<Fragment>
  <h1
    className="search__title"
  >
    FIND YOUR MOVIE
  </h1>
  <form
    className="search"
    onSubmit={[Function]}
  >
    <div
      className="search__input"
    >
      <Input
        name="search"
        onChange={[Function]}
        placeholder="What do you want to watch?"
        value=""
      />
    </div>
    <div
      className="search__button"
    >
      <Button
        isSubmit={true}
        type="primary"
      >
        SEARCH
      </Button>
    </div>
  </form>
</Fragment>
`;
